0001-short-term-release-process/0001-release-process-short-term.md

```diff
@@ -0,0 +1,325 @@
+    - Developers can deploy their own isolated testnets if necessary
+    - We can do quick `alpha` releases if necessary
+* Release candidate (RC) phase (one week):
+    - Create a `release-YYYY.MM.rc.1` branch from `main`. No new features will be accepted on this
+      branch.
+    - Bump version numbers, with an `rc.1` suffix applied.
+    - Build RC and release to Github as a public `pre-release` but with no published crates.
+    - Deploy RC to `STG-01`.
+    - Production of changelog begins.
+    - Invite community members to test against this network.
+    - Shu's metrics solution enables comparisons to be made to the previous stable release in `STG-02`.
+    - Fixes can be applied to the release branch, resulting in an `rc.2`. This will be
+      released/deployed/tested. Repeat if necessary.
+* Release and deploy phase (one/two days):
+    - On the release branch, modify version numbers to remove the `rc` pre-release specifier.
+    - Changelog finalised.
+    - Merge the release branch into `stable`.
+    - Merge the release branch into `main`.
+    - When ready, perform a release and tag of `stable` by kicking off a workflow. *
+    - Delete the release branch.
+    - Deploy to `PROD-01`/`02` and drain *
+    - Announce to users
+
+We can accommodate hotfixes at any point during the cycle.
+
+#### Uncertainties
+
+* Not completely sure about the duration of each phase. Any opinions?
+* We could deploy automatically, but do we need to coordinate with an announcement?
+* What does the 'draining' process with two production environments look like? Shu to elaborate on
+the details? Do we need this on an environment that we upgrade?
+
+## Release Cycle Anatomy
+
+We'll now elaborate the release cycle described in the last section, discussing each type of release
+in more detail.
+
+### Artifacts
+
+All release types will produce a set of binary artifacts, so we'll discuss these first. In Rust,
+crates must use Semantic Versioning. A binary is defined within a crate, and therefore, by default,
+it will also have a Semantic Version; however, it is possible to override the `--version` argument
+on the binary to provide something custom. Our releases currently produce eight binary artifacts. *
+It would be useful if we could refer to these collectively with a single version number and package,
+where the package name would reflect the version number. ** The `--version` argument would identify
+this version number, but also identify the individual component using its Semantic Version. *
+
+The proposal is to use `YYYY.MM.X.Y` as the collective version number. **
+
+We would use the collective version number for a single Github Release. The assets for the release
+would be the combined binary packages for each platform. The changelog can also be nicely applied to
+this combined release.
+
+#### Uncertainties
+
+* Should `faucet` and `sn_auditor` be part of a package targeted at users?
```

Comment:
Hmm, yes. Both could be useful in local development.

Reply:
Thanks. I've removed this as an uncertainty.